tools/castle-editor/code/framedesign.pas

```diff
@@ -1933,6 +1962,14 @@ procedure TDesignFrame.ControlsTreeEndDrag(Sender, Target: TObject; X,
   ControlsTree.Invalidate; // force custom-drawn look redraw
 end;
 
+procedure TDesignFrame.ControlsTreeKeyUp(Sender: TObject; var Key: Word;
```

Comment:
Instead of handling F2 directly, please add a a menu item, with a shortcut F2.

This way, this functionality is "discoverable" by just looking at menu items.

Reply:
I wonder where should we add this new menu item? To Design? Or maybe we should split Design into Edit, Create and Insert?

![Clipboard 1](https://user-images.githubusercontent.com/10726162/94351971-ef382e00-0067-11eb-8b77-0893f2d4d0b6.jpg)

I've added it to Design before Undo in 5e829e4874ec94db984887a6440472ce9be98994 but we might want to change it.